include/zephyr/kernel.h

```diff
@@ -6224,6 +6224,17 @@ int k_thread_runtime_stats_enable(k_tid_t thread);
  */
 int k_thread_runtime_stats_disable(k_tid_t thread);
 
+/**
+ * @brief Check if runtime statistics gathering is enabled for a thread
+ *
+ * This function checks the status of runtime statistics gathering for the
+ * specified thread and returns the current usage_lock status.
+ *
+ * @param thread ID of thread
+ * @return -EINVAL if invalid thread ID, otherwise return usage_lock status
+ */
+int k_thread_runtime_stats_is_enabled(k_tid_t thread);
```

Comment:
Is there a specific reason that this shouldn't be inline or just a compile-time check?
```suggestion
static inline int k_thread_runtime_stats_is_enabled(void)
{
    if (IS_ENABLED(CONFIG_THREAD_RUNTIME_STATS)) {
        return true;
    }

    return false;
}
```

Reply:
It seems that you are right, but I think that users can choose to turn it off or on during subsequent development, that is, it should be able to obtain the result dynamically at runtime rather than through macros determined at compile time.